plugins/operator-examples/__init__.py

```diff
@@ -815,6 +815,73 @@ def execute(self, ctx):
         ctx.ops.set_view(view=view)
 
 
+class ExampleComplexExecution(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="example_complex_execution",
+            label="Example Complex Execution",
+            allow_immediate_execution=True,
+            allow_delegated_execution=True,
+        )
+
+    def __call__(
+        self,
+        sample_collection,
+        message=None,
+        delegate=False,
+        delegation_target=None,
+    ):
+        """Illustrates a complex operation that can be run immediately or delegated.
+
+        Example usage::
+
+            import fiftyone as fo
+            import fiftyone.operators as foo
+            import fiftyone.zoo as foz
+
+            dataset = foz.load_zoo_dataset("quickstart")
+            complex_exec = foo.get_operator("@voxel51/operator-examples/example_complex_execution")
+
+            # Run immediately
+            complex_exec(dataset, message="Running immediately")
+
+            # Schedule to run on an orchestrator
+            complex_exec(dataset, delegate=True, message="Running on orchestrator")
+
+        Args:
+            delegate (False): whether to delegate execution
+            delegation_target (None): an optional orchestrator on which to
+                schedule the operation, if it is delegated
+        """
+        if isinstance(sample_collection, fo.DatasetView):
+            ctx = dict(view=sample_collection)
+        else:
+            ctx = dict(dataset=sample_collection)
+
+        if delegation_target is not None:
+            ctx["delegation_target"] = delegation_target
```

Comment:
This seems a bit hacky, would be better if it was:

```python
foo.execute_operator(delegation_target="...", ...)
```

Reply:
Agreed and what you propose is indeed valid syntax. I updated the other relevant operators to use it in https://github.com/voxel51/fiftyone-plugins/pull/183